src/WebJobs.Script/Scale/HostPerformanceManager.cs

```diff
@@ -99,5 +190,38 @@ internal ApplicationPerformanceCounters GetPerformanceCounters(ILogger logger =
 
             return null;
         }
+
+        private IFunctionInvocationDispatcher GetDispatcherAsync()
+        {
+            var hostManager = _serviceProvider.GetService<IScriptHostManager>();
+            var dispatcherFactory = (hostManager as IServiceProvider)?.GetService<IFunctionInvocationDispatcherFactory>();
+            if (dispatcherFactory != null)
```

Comment:
Why not add IFunctionInvocationDispatcherFactory as dependency in constructor?

Reply:
Tried that initially but it didn't work, due to some issues with service scopes.